scalafix-interfaces/src/main/java/scalafix/interfaces/ScalafixArguments.java

```diff
@@ -25,10 +31,45 @@
      */
     ScalafixArguments withRules(List<String> rules);
 
+    /**
+     * @param customURLs Extra URLs for classloading and compiling external rules.
+     */
+    ScalafixArguments withToolClasspath(List<URL> customURLs);
+
+    /**
+     * @param customURLs                   Extra URLs for classloading and compiling external rules.
+     * @param extraDependenciesCoordinates Extra dependencies for classloading and compiling external rules.
+     *                                     For example "com.github.liancheng::organize-imports:0.3.0".
+     *                                     Artifacts will be resolved against the Scala version in the classloader
+     *                                     of the parent {@link Scalafix} instance and fetched using Coursier.
+     * @throws ScalafixException in case of errors during artifact resolution/fetching.
+     */
+    ScalafixArguments withToolClasspath(
+            List<URL> customURLs,
+            List<String> extraDependenciesCoordinates
```

Comment:
Should this be a coursierinterface.Dependency?

Reply:
`coursierinterface.Dependency` has the Scala version hardcoded, and since that version is changing ([from 2.12 to the eventually target sources one](https://github.com/scalacenter/sbt-scalafix/pull/121)), I thought it was better to request coordinates and let the implementation append the right suffix.

Reply:
By supporting `org:::module`, the client can also easily make use of advanced community rules that are built against a full Scala version, without having to explicitly update the suffix at each patch release.

Reply:
Makes sense, good idea